.github/workflows/pre-commit.yml

```diff
@@ -35,6 +37,7 @@ jobs:
           brew install norwoodj/tap/helm-docs
       - name: pre-commit
         run: |
+          pip install pre-commit
```

Comment:
Sadly @mistercrunch this is not right. We should never run `pip install` without a specific version as it _could_ result in a non-deterministic build when running CI. Instead we should install the frozen requirements in `requirements/development.txt`. 

Granted this brings in a lot of extra packages, but this was less of an issue when we had more environments including `requirements/integration.[in|txt]` which included only the subset of packages needed to run various CI commands.

Reply:
@john-bodley maybe we should break out the light weight linting packages into a separate requirements file and depend on that in the development one?

Reply:
we can revert and load the whole bundle, not a biggy

Reply:
here -> https://github.com/apache/superset/pull/27999

Reply:
Or how about this!? https://github.com/apache/superset/pull/28000